### PR TITLE
MOEN-32219: Multiple selfhandled inapp support

### PR DESCRIPTION
**Pull request by RakshithaAcharya**
https://moengagetrial.atlassian.net/browse/MOEN-32219
Multiple selfhandled inapp support

### CODE REVIEW

File: MoEngagePluginBase/MoEngagePluginUtils.swift [hunk 1]
```diff
@@ -137,10 +127,41 @@ public class MoEngagePluginUtils {
...
+        return [MoEngagePluginConstants.InApp.screenName: rules.screenName, MoEngagePluginConstants.InApp.contexts: rules.contexts]
+    }
+
+    static func mapSelfHandledCampaignToJSON(_ campaign: MoEngageInAppSelfHandledCampaign) -> [String: Any] {
+        var campaignPayload = campaign.fetchInAppPaylaod()
+      
+        var selfHandledPayload = [String: Any]()
+        selfHandledPayload[MoEngagePluginConstants.General.payload] = campaign.campaignContent
+        selfHandledPayload[MoEngagePluginConstants.InApp.dismissInterval] = campaign.autoDismissInterval
+        selfHandledPayload[MoEngagePluginConstants.InApp.displayRules] = mapDisplayRulesToJSON(campaign.displayRules)
```
Comment (babulmoengage) [inline MoEngagePluginBase/MoEngagePluginUtils.swift:158]: Is it fine to add display rules for normal self handled campaigns  also ?

Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePluginUtils.swift:158]: @babulmoengage its added for existing self handled api

Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePluginUtils.swift:158]: @babulmoengage its added for existing self handled api. This is common code

Comment (RakshithaAcharya) [reply MoEngagePluginBase/MoEngagePluginUtils.swift:158]: @babulmoengage its added for existing self handled api. This is common code

**Review by babulmoengage — APPROVED** [review]

**Pull request merged**